Add Intermediate Haskell link

ocramz opened the pull request:
As promised this morning at Haskell Exchange :)

ocramz committed:
Add Intermediate Haskell

ocramz committed:
Add "Haskell study syllabus"

sassela commented:
Thanks so much for this @ocramz! 

1) I'm unsure about adding a link to Intermediate Haskell, as it doesn't look like there's any Haskell learning material publicly available yet. Please remove it for now and I will definitely reconsider adding it once the authors have made the resource public.

2) The Haskell study syllabus looks more like an intermediate meta-resource to me, ie. a page containing links to other resources. If you agree, please could you put it under that "Intermediate meta-resource" heading?

Thanks!

ocramz committed:
Applied feedback

Deleted the "Intermediate Haskell" landing page, moved the FPCo Syllabus to meta-resources section

ocramz commented:
You're right, I've applied your suggestions :) However the FPComplete syllabus actually contains a number of good systems programming points in its own right.

sassela commented:
That's a fair point. Thanks for this, all merged
